Microsoft.Azure.Cosmos/src/CosmosClientOptions.cs

```diff
@@ -633,6 +635,17 @@ internal Protocol ConnectionProtocol
         /// </summary>
         internal Func<TransportClient, TransportClient> TransportClientHandlerFactory { get; set; }
 
+        /// <summary>
+        /// A callback delegate to validate the server certificate.
+        /// </summary>
+        /// <remarks>
+        /// <para>
+        /// This is applicable when client service has issue connecting to server using IP and needs certificate validation.
+        /// This is the contract available to client which would override RemoteCertificateValidationCallback for TCP and ServerCertificateCustomValidationCallback for HTTP.
+        /// </para>
+        /// </remarks>
+        public Func<object, X509Certificate2, X509Chain, SslPolicyErrors, bool> ServerCertificateCustomValidationCallback { get; set; }
```

Comment:
The Azure SDK version does not have the object/HttpRequestMessage:

```
public ServerCertificateCustomValidationArgs(X509Certificate2? certificate, X509Chain? certificateAuthorityChain, SslPolicyErrors sslPolicyErrors)
```

Can it be simply:

```suggestion
        public Func<X509Certificate2, X509Chain, SslPolicyErrors, bool> ServerCertificateCustomValidationCallback { get; set; }
```

And for HTTP, we can simply do like https://github.com/Azure/azure-sdk-for-net/blob/4f54c9cb440c9db05d26b0538ff3bce2bbc3c82d/sdk/core/Azure.Core/src/Pipeline/HttpClientTransport.cs#L565

```
if (serverCertificateCustomValidationCallback != null)
{
	httpHandler.ServerCertificateCustomValidationCallback = (_, certificate2, x509Chain, sslPolicyErrors) =>
	{
		return serverCertificateCustomValidationCallback(certificate2, x509Chain, sslPolicyErrors));
	};
}
```

Reply:
It will give errors as RemoteCertificateValidationCallback has object parameter.

public delegate bool RemoteCertificateValidationCallback(object sender, X509Certificate certificate, X509Chain chain, SslPolicyErrors sslPolicyErrors);

Reply:
updated